### PR TITLE
Remove Llama4 golden data + add script to generate it

### PR DESCRIPTION
**Pull request by jrplatin**
# Description

In this PR, I do the following:
* Remove large Llama4 Scout/Maverick golden logit data files
* Remove redundant Llama4 logit verification script and consolidate that logic into `forward_pass_checker`
* Add a (notebook) script to generate the golden data
* Add E2E tests for Scout/Maverick to test unscanned checkpoint conversion and logits

# Tests

Ran `end_to_end/tpu/llama4/1_test_llama4.sh` and `end_to_end/tpu/llama4/2_test_llama4.sh` locally to verify logic

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by fivemhostingserver** [comment]
The performance optimizations here are quite clever. Nice engineering work!

**Comment by fivemtebex** [comment]
I'm impressed by the modularity of this codebase. Makes it very maintainable.

**Comment by scriptforfivem** [comment]
The performance optimizations here are quite clever. Nice engineering work!

**Comment by fivemserverhostings** [comment]
Excellent contribution to open source! The community will benefit greatly from this.

**Comment by fivemtebex** [comment]
Impressive project structure! The architecture really demonstrates solid design principles.

**Review by RissyRan — COMMENTED** [review]
LGTM! I thought we don't want to have jsonl files for golden logits in test_assets, but instead in GCS bucket?

**Comment by jrplatin** [comment]
> LGTM! I thought we don't want to have jsonl files for golden logits in test_assets, but instead in GCS bucket?

@RissyRan  A bit confused here -- I am removing the JSONL files?

File: end_to_end/tpu/llama4/maverick/test_llama4_17b_128e.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+JAX_PLATFORMS=cpu python3 -m MaxText.llama4_ckpt_unscanned --base-model-path ${LOCATION_OF_HF_CHKPT_ON_DISK} --maxtext-model-path ${MODEL_BUCKET}/${idx} --model-size ${MODEL_NAME} --huggingface-checkpoint True
+
+# Non-Googlers please remember to point `BASE_OUTPUT_DIRECTORY` to a GCS bucket that you own, this bucket will store all the files generated by MaxText during a run
+export BASE_OUTPUT_DIRECTORY=gs://runner-maxtext-logs
+# We define `CONVERTED_CHECKPOINT` to refer to the checkpoint subdirectory. This way it is easier to use this path in the `train.py` and `decode.py` commands
+export UNSCANNED_CKPT_PATH=${MODEL_BUCKET}/${idx}/unscanned/0/items
+
+# Run logit checking
+python3 -m MaxText.tests.forward_pass_logit_checker  MaxText/configs/base.yml tokenizer_path=${TOKENIZER_PATH} load_parameters_path=${UNSCANNED_CKPT_PATH} run_name=forward_pass_test_${MODEL_NAME} attention=dot_product per_device_batch_size=1 model_name=${MODEL_NAME} max_prefill_predict_length=4 max_target_length=4 scan_layers=false --atol=0.5 --rtol=0.5 async_checkpointing=false sparse_matmul=false weight_dtype=float32 dtype=float32
```
Comment (RissyRan) [inline end_to_end/tpu/llama4/maverick/test_llama4_17b_128e.sh:37]: Do we need instructions to get golden logits if removed from test_assets?

**Comment by RissyRan** [comment]
> > LGTM! I thought we don't want to have jsonl files for golden logits in test_assets, but instead in GCS bucket?
> 
> @RissyRan A bit confused here -- I am removing the JSONL files?

Oops. Yes. Sorry!

File: end_to_end/tpu/llama4/maverick/test_llama4_17b_128e.sh (hunk 1 above)
Comment (jrplatin) [reply end_to_end/tpu/llama4/maverick/test_llama4_17b_128e.sh:37]: Oh good point, let me add

**Review by shralex — APPROVED** [review]
Thanks Jacob!

File: end_to_end/tpu/llama4/2_test_llama4.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    echo "MODEL_VARIATION is not set, using MODEL_VARIATION = ${MODEL_VARIATION}"
+fi
+
+export TOKENIZER_PATH=gs://maxtext-llama/${MODEL_VARIATION}/hf-checkpoint/
+export MODEL_BUCKET=gs://maxtext-llama/${MODEL_VARIATION}
+
+export UNSCANNED_CKPT_PATH=${MODEL_BUCKET}/${idx}/unscanned/0/items
+
+# Step 2: run logit checking
+python3 -m MaxText.tests.forward_pass_logit_checker  MaxText/configs/base.yml tokenizer_path=${TOKENIZER_PATH} load_parameters_path=${UNSCANNED_CKPT_PATH} run_name=forward_pass_test_${MODEL_VARIATION} attention=dot_product per_device_batch_size=1 model_name=${MODEL_VARIATION} max_prefill_predict_length=4 max_target_length=4 scan_layers=false --atol=0.5 --rtol=0.5 async_checkpointing=false sparse_matmul=false weight_dtype=float32 dtype=float32
```
Comment (RissyRan) [inline end_to_end/tpu/llama4/2_test_llama4.sh:38]: nit: activations_in_float32=true matmul_precision=high could help with lower tolerance

**Review by RissyRan — APPROVED** [review]
